qiskit/ignis/verification/randomized_benchmarking/dihedral.py

```diff
@@ -338,15 +336,22 @@ class CNOTDihedral():
            npj Quantum Inf 2, 16012 (2016).
     """
 
-    def __init__(self, num_qubits):
+    def __init__(self, data):
```

Comment:
To fit in with the operator classes the init function should be able to initialized from the raw data structures used, or convert from other operator / circuit types. Looking at this method the data structures are linear, poly, shift so you could do something like:
```
__init__(data, shift=None, poly=None)
```
with data checked on
- if np.array it is treated as the n x n linear matrix and check for poly and linear
- if circuit / instruction initialize using the from_instruction
- if ScalarOp initialize an identity
- if Pauli or Clifford initialize by converting those operators to CNOTDihedral (if you know how to in general)

Maybe you don't want to make `poly` and `shift` kwargs though, and just have a single data arg

Reply:
The CNOT-Dihedral group does not contain the Clifford group (since it does not contain the H gate), it only contains the Pauli group. In fact, CNOT-Dihedral + Clifford is a universal gateset.

Reply:
Added initialization from:
- CNOTDihedral
- ScalarOp
- QuantumCircuit
- Instruction
- Pauli